src/common/dsp/modulators/FormulaModulationHelper.cpp

```diff
@@ -810,13 +820,26 @@ std::vector<DebugRow> createDebugDataOfModState(const EvaluatorState &es)
         }
     };
 
-    rec(0, false);
-    lua_pop(es.L, -1);
+    std::vector<std::string> tablesList = {es.stateName, "shared"};
```

Comment:
So `es.stateName` exists so each modulator gets a unique state. "shared" allows them to be shared. But I just want to confirm - that's shared within the LUA state, so it's not a cross-instance global right?

Also since we use the string "shared" in several places, I wonder if we want a constexpr char* for it which we use rather than a quoted string.

Very nice moving this to a loop here by the way. Diff confused me at first but I like this approach way better.

Reply:
"shared" is the name of an empty table that gets added [here](https://github.com/nuoun/surge/blob/1ee52c6a8317ff7f31cbc4d59e72bff41a432571/src/common/dsp/modulators/FormulaModulationHelper.cpp#L84) the same way the "surge" table is added that is used to load in the prelude. Since this gets added on the Lua state just once using that `if (firstTimeThrough)` block and before the sandbox environment and evaluation state gets added it is accessible for every Lua Formulator instance in voice(s) and scenes (IIRC only the unittests use their own Lua state).

I wanted to call it "global" initially but while that is not a keyword it has a bunch of uses in Lua so figured this might be less confusing. Both setglobal() and getglobal() use `const char *name` so I suppose we could use a constexpr char* for that vector too I suppose.

Reply:
Yeah I have no problem with shared. I just meant lets add

```
static constexpr const char* surgeTableName{"surge"};
static constexpr const char* sharedTableName("shared"};
```

somewhere and not use string literals. Was a minor comment to that end.